packit/utils/changelog_helper.py

```diff
@@ -90,9 +90,14 @@ def get_entry_from_action(
 
     @staticmethod
     def sanitize_entry(entry: str) -> str:
-        # escape macro references and macro/shell/expression expansions
-        # that could break spec file parsing
-        entry = re.sub(r"(?<!%)%(?=(\w+|[{[(]))", "%%", entry)
+        # 1. split the entry by '%%' to protect already sanitized percent sign occurrences
+        # 2. in each part escape macro references and macro/shell/expression expansions
+        #    that could break spec file parsing
+        # 3. join parts by '%%' again
+        entry = "%%".join(
+            [re.sub(r"%(?=(\w+|[{[(]))", "%%", part) for part in entry.split("%%")],
+        )
```

Comment:
To be 100% sure we could use `ValueParser` from specfile to parse, escape and reconstruct the string. Example of use:
https://github.com/packit/specfile/blob/61e261ca01cc91291cb501e3fda3867379442a71/specfile/specfile.py#L557-L562
Docs are here: https://packit.dev/specfile/api/specfile/value_parser

Reply:
Something like this should work:

```python
def escape_macros(nodes):
    result = ""
    for node in nodes:
        if isinstance(node, ConditionalMacroExpansion):
            result += f"%%{{{node.prefix}{node.name}:"
            result += escape_macros(node.body)
            result += "}"
        elif isinstance(node, StringLiteral):
            result += str(node)
        else:
            result += "%" + str(node)
    return result

return escape_macros(ValueParser.parse(entry))
```

Reply:
Thanks.